pkg/components/linkerd/component.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2020 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package linkerd
+
+// nolint: gci
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/gohcl"
+	"github.com/linkerd/linkerd2/pkg/tls"
+
+	internaltemplate "github.com/kinvolk/lokomotive/internal/template"
+	"github.com/kinvolk/lokomotive/pkg/components"
+	"github.com/kinvolk/lokomotive/pkg/components/util"
+	"github.com/kinvolk/lokomotive/pkg/k8sutil"
+)
+
+const (
+	name = "experimental-linkerd"
+)
+
+// nolint:gochecknoinits
+func init() {
+	components.Register(name, newComponent())
+}
+
+type component struct {
+	ControllerReplicas int `hcl:"controller_replicas,optional"`
+
+	CA     string
+	Cert   string
+	Key    string
+	Expiry string
+}
+
+func newComponent() *component {
+	return &component{
+		ControllerReplicas: 1,
+	}
+}
+
+func (c *component) LoadConfig(configBody *hcl.Body, evalContext *hcl.EvalContext) hcl.Diagnostics {
+	diagnostics := hcl.Diagnostics{}
+
+	if configBody == nil {
+		return hcl.Diagnostics{}
+	}
+
+	d := gohcl.DecodeBody(*configBody, evalContext, c)
+	if d.HasErrors() {
+		return append(diagnostics, d...)
+	}
+
+	return diagnostics
+}
+
+func (c *component) RenderManifests() (map[string]string, error) {
+	helmChart, err := util.LoadChartFromAssets("/components/linkerd2")
```

Comment:
This "2" is annoying. Can we move assets to `components/linkerd`?

Reply:
I don't know, everytime we do a fetch there will be additional step to rename the dir. Hence I kept the default.

Reply:
Right. I guess it's fine then. Can we have a comment here, that `linkerd2` is a name of the upstream chart please?